### PR TITLE
[RPKG] Fix packagedef/thumbs XTEA encrypt/decrypt

### PR DESCRIPTION
**Pull request by AnthonyFuller**
This fixes the encrypting/decrypting function of `encrypt_packagedefinition_thumbs` and `decrypt_packagedefinition_thumbs` respectively. Also removes the need of meta files for encrypting.

### CODE REVIEW

File: rpkg_src/decrypt_packagedefinition_thumbs.cpp [hunk 1]
```diff
@@ -21,16 +22,21 @@ void generic_function::decrypt_packagedefinition_thumbs(std::string& input_path,
 
     uint64_t packagedefinitions_thumbs_file_size = file.tellg();
 
-    int packagedefinitions_thumbs_header_size = 20;
+    int packagedefinitions_thumbs_header_size = 16;
 
-    packagedefinitions_thumbs_file_size -= packagedefinitions_thumbs_header_size;
+	packagedefinitions_thumbs_file_size -= packagedefinitions_thumbs_header_size;
+	packagedefinitions_thumbs_file_size -= 4; // -4 to skip the checksum
```
Comment (RDIL) [inline rpkg_src/decrypt_packagedefinition_thumbs.cpp:28]:
```suggestion
    packagedefinitions_thumbs_file_size -= packagedefinitions_thumbs_header_size;
    packagedefinitions_thumbs_file_size -= 4; // -4 to skip the checksum
```

File: rpkg_src/decrypt_packagedefinition_thumbs.cpp [hunk 2]
```diff
@@ -21,16 +22,21 @@ void generic_function::decrypt_packagedefinition_thumbs(std::string& input_path,
...
     file.seekg(0, file.beg);
 
     std::vector<char> packagedefinitions_thumbs_header(packagedefinitions_thumbs_header_size, 0);
 
     file.read(packagedefinitions_thumbs_header.data(), packagedefinitions_thumbs_header_size);
 
+	std::vector<char> checksum_raw(4, 0);
+
+	file.read(checksum_raw.data(), 4);
+
```
Comment (RDIL) [inline rpkg_src/decrypt_packagedefinition_thumbs.cpp:39]:
```suggestion
    std::vector<char> checksum_raw(4, 0);

    file.read(checksum_raw.data(), 4);

```

File: rpkg_src/decrypt_packagedefinition_thumbs.cpp [hunk 3]
```diff
@@ -73,7 +79,18 @@ void generic_function::decrypt_packagedefinition_thumbs(std::string& input_path,
...
+	uint32_t crc = crc32::update(table, 0, input_data.data(), last_zero_position);
+
+	uint32_t old_crc = 0;
+	memcpy(&old_crc, &checksum_raw.data()[0], sizeof(old_crc));
+
+	std::string output_file_base_name = file::output_path_append(file::get_base_file_name(input_path), output_path);
+
+	if (crc != old_crc) {
+		LOG_AND_EXIT("Could not decrypt " + output_file_base_name + "!\nReason: Checksum mismatch! Exiting...");
+	}
```
Comment (RDIL) [inline rpkg_src/decrypt_packagedefinition_thumbs.cpp:93]:
```suggestion
    uint32_t table[256];
    crc32::generate_table(table);
    uint32_t crc = crc32::update(table, 0, input_data.data(), last_zero_position);

    uint32_t old_crc = 0;
    memcpy(&old_crc, &checksum_raw.data()[0], sizeof(old_crc));

    std::string output_file_base_name = file::output_path_append(file::get_base_file_name(input_path), output_path);

    if (crc != old_crc) {
        LOG_AND_EXIT("Could not decrypt " + output_file_base_name + "!\nReason: Checksum mismatch! Exiting...");
	}
```

**Review by RDIL — CHANGES_REQUESTED** [review]
Tony you're killing me with these indents

File: rpkg_src/decrypt_packagedefinition_thumbs.cpp (hunk 3 above)
Comment (AnthonyFuller) [reply rpkg_src/decrypt_packagedefinition_thumbs.cpp]: Blame Visual Studio. Fixed in another commit.

File: rpkg_src/decrypt_packagedefinition_thumbs.cpp (hunk 2 above)
Comment (AnthonyFuller) [reply rpkg_src/decrypt_packagedefinition_thumbs.cpp]: Blame Visual Studio. Fixed in another commit.

File: rpkg_src/decrypt_packagedefinition_thumbs.cpp (hunk 1 above)
Comment (AnthonyFuller) [reply rpkg_src/decrypt_packagedefinition_thumbs.cpp]: Blame Visual Studio. Fixed in another commit.

**Review by RDIL — APPROVED** [review]

**Pull request merged**